src/hook.js

```diff
@@ -155,6 +157,27 @@ export function installHook(target: any): DevToolsHook | null {
       ? 'deadcode'
       : detectReactBuildType(renderer);
 
+    // Patching the console enables DevTools to do a few useful things:
+    // * Append component stacks to warnings and error messages
+    // * Disable logging during re-renders to inspect hooks (see inspectHooksOfFiber)
+    //
+    // For React Native, we intentionally patch early (during injection).
+    // This provides React Native developers with components stacks even if they don't run DevTools.
+    // This won't work for DOM though, since this entire file is eval'ed and inserted as a script tag.
+    // In that case, we'll patch later (when the frontend attaches).
+    //
+    // Don't patch in test environments because we don't want to interfere with Jest's own console overrides.
+    if (process.env.NODE_ENV !== 'test') {
+      try {
+        // We try/catch this instead of doing a typeof check,
+        // because Webpack generates JS output that will runtime error with the typeof check.
+        // e.g. typeof patchConsole === 'function'
+        // becomes: typeof _backend_console__WEBPACK_IMPORTED_MODULE_0__["patch"] === 'function'
+        // and _backend_console__WEBPACK_IMPORTED_MODULE_0__ itself is undefined
```

Comment:
Why is it undefined? I don't think I understand this.

Reply:
There's a comment in the header about this too, but basically for the browser version- we inject the hook by to-stringing it.

https://github.com/bvaughn/react-devtools-experimental/blob/0f2fb5badf5908ee271e0b9e7ec7f3022e31a648/shells/browser/shared/src/injectGlobalHook.js#L85-L89

Because of this, imports won't work- although they will for React Native.

If you have suggestions on the wording I would be happy to clean it up.

Reply:
I pushed a wording improvement.